src/cpp/link_creation_agent/agent.cc

```diff
@@ -0,0 +1,83 @@
+#include "agent.h"
+#include "RemoteIterator.h"
+
+#include <fstream>
+#include <sstream>
+
+using namespace std;
+using namespace link_creation_agent;
+using namespace query_node;
+using namespace query_element;
+
+LinkCreationAgent::LinkCreationAgent(string config_path) {
+    this->config_path = config_path;
+    load_config();
+    link_creation_node_server = new LinkCreationNode(link_creation_server_id);
+    query_node_client = new DASNode(query_node_client_id, query_node_server_id);
+
+    this->agent_thread = new thread(&LinkCreationAgent::run, this);
+}
+
+LinkCreationAgent::~LinkCreationAgent() {
+    stop();
+    delete link_creation_node_server;
+}
+
+void LinkCreationAgent::stop() {
+    link_creation_node_server->graceful_shutdown();
+    if (agent_thread != NULL && agent_thread->joinable()) {
+        agent_thread->join();
+        agent_thread = NULL;
+    }
+}
+
+void LinkCreationAgent::run() {
+    while (true) {
+        this_thread::sleep_for(chrono::seconds(loop_interval));
```

Comment:
The thread is supposed to sleep only when there's no further requests to process. Something like:

```
while true
    if (request queue is empty)
        then sleep
        else process request
```